src/test/java/io/confluent/copycat/hdfs/HdfsWriterTest.java

```diff
@@ -123,19 +124,46 @@ public void testWriteRecordNonZeroInitailOffset() throws IOException, CopycatExc
 
     long[] validOffsets = {2, 5, 8, 9};
     for (int i = 1; i < validOffsets.length; i++) {
-      Path path = new Path(FileUtils.committedFileName(url, topicsDir, topicPart, validOffsets[i]));
+      long startOffset = validOffsets[i - 1] + 1;
+      long endOffset = validOffsets[i];
+      Path
+          path =
+          new Path(FileUtils.committedFileName(url, topicsDir, topicPart, startOffset, endOffset));
       Collection<Object> records = readAvroFile(path);
-      long size = validOffsets[i] - validOffsets[i - 1];
+      long size = endOffset - startOffset + 1;
       assertEquals(records.size(), size);
-      for (Object avroRecord: records) {
+      for (Object avroRecord : records) {
         assertEquals(avroRecord, AvroData.convertToAvro(record));
       }
     }
+  }
+
+  @Test
+  public void testTimedRotate() throws IOException, CopycatException, InterruptedException {
+    Properties props = createProps();
+    HdfsSinkConnectorConfig connectorConfig = new HdfsSinkConnectorConfig(props);
+    HdfsWriter hdfsWriter = new HdfsWriter(connectorConfig);
+
+    String topicsDir = connectorConfig.getString(HdfsSinkConnectorConfig.TOPIC_DIR_CONFIG);
+    String topic = "topic";
+    int partition = 0;
+    String key = "key";
+    GenericRecord record = createRecord();
+    TopicPartition topicPart = new TopicPartition(topic, partition);
+
+    for (long offset = 0; offset < 2; offset++) {
+      SinkRecord sinkRecord = new SinkRecord(topic, partition, key, record, offset);
+      hdfsWriter.writeRecord(topicPart, sinkRecord);
+    }
+
+    Thread.sleep(11000);
+    hdfsWriter.close();
+
+    Path path = new Path(FileUtils.committedFileName(url, topicsDir, topicPart, 0, 1));
+    Collection<Object> records = readAvroFile(path);
 
-    long[] inValidOffsets = {3, 4, 6, 7};
-    for (long offset: inValidOffsets) {
-      Path path = new Path(FileUtils.committedFileName(url, topicsDir, topicPart, offset));
-      assertFalse(fs.exists(path));
+    for (Object avroRecord : records) {
+      assertEquals(avroRecord, AvroData.convertToAvro(record));
```

Comment:
It would be good to also test:
1. Writing for multiple partitions, making sure we get the right output in the right places.
2. Simulate certain recovery modes by setting up the files to be in the right state before creating the HdfsWriter. It should be able to recover in the expected way from those errors.
3. Any way to introduce errors in HDFS to simulate certain types of errors? e.g. anything that might cause a transient issue, like a loss of connectivity to HDFS?
4. Multiple users of the WAL file at once? (Maybe this is better placed in a WALTest to accompany the WAL class rather than as part of HdfsWriter tests.) For example, if another process is currently holding the WAL lease, do we handle that properly?